test/integration/evictions/evictions_test.go

```diff
@@ -161,76 +163,86 @@ func TestConcurrentEvictionRequests(t *testing.T) {
 		}
 	}
 
-	waitToObservePods(t, podInformer, 10)
+	waitToObservePods(t, podInformer, numOfEvictions)
 
 	pdb := newPDB()
 	if _, err := clientSet.Policy().PodDisruptionBudgets(ns.Name).Create(pdb); err != nil {
 		t.Errorf("Failed to create PodDisruptionBudget: %v", err)
 	}
 
-	waitPDBStable(t, clientSet, 10, ns.Name, pdb.Name)
+	waitPDBStable(t, clientSet, numOfEvictions, ns.Name, pdb.Name)
 
-	doneCh := make(chan bool, 10)
-	errCh := make(chan error, 1)
-	// spawn 10 goroutine to concurrently evict the pods
-	for i := 0; i < 10; i++ {
+	doneCh := make(chan bool, numOfEvictions)
+	errCh := make(chan error, 2*numOfEvictions)
+	var wg sync.WaitGroup
+	// spawn numOfEvictions goroutines to concurrently evict the pods
+	for i := 0; i < numOfEvictions; i++ {
+		wg.Add(1)
 		go func(id int, doneCh chan bool, errCh chan error) {
-			evictionName := fmt.Sprintf(podNameFormat, id)
-			eviction := newEviction(ns.Name, evictionName, deleteOption)
-
+			defer wg.Done()
+			podName := fmt.Sprintf(podNameFormat, id)
+			eviction := newEviction(ns.Name, podName, deleteOption)
 			var e error
-			for {
+			if err := wait.PollImmediate(5*time.Second, 60*time.Second, func() (bool, error) {
```

Comment:
due to the length of this conditional block (due to the length of the anonymous function) it might be a little easier to follow along if we assign `err` on line 186 and send the error over `errCh` if it's non `nil` and then we can bring the lines in the `else` block if we return out of the goroutine after sending a non `nil` error over `errCh`